### PR TITLE
Changed repository dir to use the PR source directory. Fixes #72

### LINKED ISSUE #72: Merge check hooks can not access PR target reference
When a script is configured as a merge check hook, the target reference commit (to_ref as provided on STDIN) of the pull request is not available to the script.

To reproduce:

* Create a script with the following content:
```
while read from_ref to_ref ref_name; do
    echo "Ref update:"
    echo " Old value: $from_ref"
    echo " New value: $to_ref"
    echo " Ref name:  $ref_name"
    echo " Diff:"
    git show $from_ref..$to_ref | sed 's/^/  /'
done
exit 1 # don't allow a merge just yet as we're debugging. ;-)
```
* Enable this script in a repository merge check hook.
* Fork the repository
* Clone, commit and push something in the repository
* Create a pull request for your change

Expected result:

* The script shows the differences in the pull request and exit with rc 1

Actual Result:

* The script exits with a `bad object` error

Tested on:

* Bitbucket Server 5.10.1, GIT 2.7.4
* Bitbucket Server 5.10.1, GIT 2.17.1

Problem:

Push hooks have GIT_* environment variables available. These variables, specifically GIT_OBJECT_DIRECTORY and GIT_ALTERNATE_OBJECT_DIRECTORIES, combine the pull request target repository objects with the pull request source repository objects and allow to access the commits from the pull request.

These are not available when using a merge check hook. Trying to add the `@SynchronousPreferred` annotation didn't help.

Workaround:

The following script is a rather hacky and ugly workaround:

```
if [ "${PULL_REQUEST_FROM_REPO_ID}" ]
then
    # This is a pull request. Call git show from the originating repository
    cd ../${PULL_REQUEST_FROM_REPO_ID}
fi
while read from_ref to_ref ref_name; do
    echo "Ref update:"
    echo " Old value: $from_ref"
    echo " New value: $to_ref"
    echo " Ref name:  $ref_name"
    echo " Diff:"
    git show $from_ref..$to_ref | sed 's/^/  /'
done
exit 1 # Still debugging first
```

### PR DESCRIPTION
**Pull request by dploeger**

### CODE REVIEW

File: src/main/java/com/ngs/stash/externalhooks/hook/ExternalMergeCheckHook.java [hunk 1]
```diff
@@ -155,7 +155,7 @@ public RepositoryHookResult preUpdate(PreRepositoryHookContext context, PullRequ
             }
         }
 
-        Repository repo = pr.getToRef().getRepository();
+        Repository repo = pr.getFromRef().getRepository();
```
Comment (seletskiy) [inline src/main/java/com/ngs/stash/externalhooks/hook/ExternalMergeCheckHook.java]: Isn't that going to change result of the following calls — [ExternalPreReceiveHook.java#L146-L150](https://github.com/reconquest/atlassian-external-hooks/blob/master/src/main/java/com/ngs/stash/externalhooks/hook/ExternalPreReceiveHook.java#L146-L150)?

Comment (dploeger) [reply src/main/java/com/ngs/stash/externalhooks/hook/ExternalMergeCheckHook.java]: Oh, right. Didn't see that.

Comment (dploeger) [reply src/main/java/com/ngs/stash/externalhooks/hook/ExternalMergeCheckHook.java]: We definitely need some test suite! 😄

Comment (dploeger) [reply src/main/java/com/ngs/stash/externalhooks/hook/ExternalMergeCheckHook.java]: Updated to only change the repopath

**Pull request merged**